chirp/drivers/icr6.py

```diff
@@ -0,0 +1,794 @@
+  u8 unknown03;  // 6bd3
+  u8 unknown04;  // 6bd4
+  u8 unknown05;  // 6bd5
+  u8 unknown06;  // 6bd6
+  u8 unknown07;  // 6bd7
+  u8 unknown08;  // 6bd8
+  u8 unknown09;  // 6bd9
+  u8 unknown10;  // 6bda
+  u8 unknown11;  // 6bdb
+  u8 unknown12;  // 6bdc
+  u8 unknown13_6bdd:6,
+     func_dial_step:2;    // 00=100kHz, 01=1MHz, 02=10MHz
+  u8 unknown14;  // 6bde
+  u8 unknown15_6bdf:7,
+     key_beep:1;          // 0=Off, 1=On
+  u8 unknown16_6be0:2,
+     beep_level:6;        // 0x00=Volume, 0x01=00, 0x02=01, ..., 0x28=39
+  u8 unknown17_6be1:6,
+     back_light:2;        // 00=Off, 01=On, 02=Auto1, 03=Auto2
+  u8 unknown18_6be2:7,
+     power_save:1;        // 0=Off, 1=On
+  u8 unknown17_6be3:7,
+     am_ant:1;           // 0=Ext, 1=Bar
+  u8 unknown20_6be4:7,
+     fm_ant:1;           // 0=Ext, 1=Ear (headset lead)
+  u8 unknown21;  // 6be5
+  u8 unknown22;  // 6be6
+  u8 unknown23;  // 6be7
+  u8 unknown24;  // 6be8
+  u8 unknown25;  // 6be9
+  u8 unknown26;  // 6bea
+  u8 unknown27;  // 6beb
+  u8 unknown28;  // 6bec
+  u8 unknown29;  // 6bed
+  u8 unknown30;  // 6bee
+  u8 unknown31;  // 6bef
+  u8 unknown32;  // 6bf0
+  u8 unknown33;  // 6bf1
+  u8 civ_address;         // 6bf2: CI-V address, full byte
+  u8 unknown35_6bf3:5,
+     civ_baud_rate:3;     // 6bf3: Index to CIV_BAUD_RATES, range 0-5
+  u8 unknown35_6bf4:7,
+    civ_transceive:1;    // 6bf4: Report frequency and mode changes
+  u8 unknown37;  // 6bf5
+  u8 unknown38;  // 6bf6
+  u8 unknown39;  // 6bf7
+  u8 unknown40;  // 6bf8
+  u8 unknown41;  // 6bf9
+  u8 unknown42;  // 6bfa
+  u8 unknown43;  // 6bfb
+  u8 unknown44;  // 6bfc
+  u8 unknown45;  // 6bfd
+  u8 unknown46;  // 6bfe
+  u8 unknown47;  // 6bff
+  u8 unknown48;  // 6c00
+  u8 unknown49;  // 6c01
+  u8 unknown50;  // c602
+  u8 unknown51;  // c603
+  u8 unknown52h_6c04:3,   // Fixed 001 seen during tests
+     dial_function:1,     // 0=Tuning Dial, 1=Audio Volume
+     unknown52m_6c04:2,   // Fixed 10 seen during tests
+     mem_display_type:2;  // 00=Freq, 01=BankName, 02=MemName, 03=ChNum
+  u8 unknown54;  // 6c05
+  u8 unknown55;  // 6c06
+  u8 unknown56;  // 6c07
+  u8 unknown57;  // 6c08
+  u8 unknown58;  // 6c09
+  u8 unknown59;  // 6c0a
+  u8 unknown60;  // 6c0b
+  u8 unknown61;  // 6c0c
+  u8 unknown62;  // 6c0d
+  u8 unknown63;  // 6c0e
+  u8 unknown53;  // 6c0f
+} settings;
+
+
+// Unknown. Coded as 15 blocks of 16-bytes. 0x6C10 to 0x6CFF
+struct {
+  u8 unknown0;
+  u8 unknown1;
+  u8 unknown2;
+  u8 unknown3;
+  u8 unknown4;
+  u8 unknown5;
+  u8 unknown6;
+  u8 unknown7;
+  u8 unknown8;
+  u8 unknown9;
+  u8 unknown10;
+  u8 unknown11;
+  u8 unknown12;
+  u8 unknown13;
+  u8 unknown14;
+  u8 unknown15;
+} mystery4[15];
+
+// Device comment string. Grab it from the ICF?
+struct { // Start: 6D00, End: 6D0F
+  char comment[16];
+} device_comment;
+
+// 22x ASCII-coded bank names
+// 0x6D10 to 0x6DBF inclusive
+struct {
+  char name[6];
+  u8 unknown0; // Padding?
+  u8 unknown1; // Padding?
```

Comment:
`u8 padding[2]` will save a line in things like this.

Reply:
Fixed